src/compiler/checker.ts

```diff
@@ -18595,7 +18595,19 @@ namespace ts {
             // Grammar checking
             checkGrammarStatementInAmbientContext(node);
 
-            checkExpression(node.expression);
+            const type = checkExpression(node.expression);

```

Comment:
It is not possible to have nested promises. A promise resolved with a promise simply resolves with the same completion value as the "inner" promise.

`await await bar()` will first unwrap the Promise that `bar` returns, and then just await on a non-Promise, which just means queueing yourself at the end of the current micro-task queue.